java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -143,22 +144,24 @@ private void save(UnitTestIndex index, SensorContext context) {
     }
   }
 
-  private void save(UnitTestClassReport report, Resource resource, SensorContext context) {
-    double testsCount = report.getTests() - report.getSkipped();
-    saveMeasure(context, resource, CoreMetrics.SKIPPED_TESTS, report.getSkipped());
-    saveMeasure(context, resource, CoreMetrics.TESTS, testsCount);
-    saveMeasure(context, resource, CoreMetrics.TEST_ERRORS, report.getErrors());
-    saveMeasure(context, resource, CoreMetrics.TEST_FAILURES, report.getFailures());
-    saveMeasure(context, resource, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
+  private void save(UnitTestClassReport report, InputFile inputFile, SensorContext context) {
+    int testsCount = report.getTests() - report.getSkipped();
+    saveMeasure(context, inputFile, CoreMetrics.SKIPPED_TESTS, report.getSkipped());
+    saveMeasure(context, inputFile, CoreMetrics.TESTS, testsCount);
+    saveMeasure(context, inputFile, CoreMetrics.TEST_ERRORS, report.getErrors());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_FAILURES, report.getFailures());
+    saveMeasure(context, inputFile, CoreMetrics.TEST_EXECUTION_TIME, report.getDurationMilliseconds());
     double passedTests = testsCount - report.getErrors() - report.getFailures();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Cast one of the operands of this substraction operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2184)